src/main/java/org/sagebionetworks/bridge/sdk/models/upload/UploadFieldDefinition.java

```diff
@@ -70,40 +121,113 @@ public UploadFieldType getType() {
         return type;
     }
 
+    /** {@inheritDoc} */
     @Override
-    public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + Objects.hashCode(name);
-        result = prime * result + Objects.hashCode(required);
-        result = prime * result + Objects.hashCode(type);
-        return result;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
+    public boolean equals(Object o) {
+        if (this == o) {
             return true;
         }
-        if (obj == null || getClass() != obj.getClass()) {
+        if (o == null || getClass() != o.getClass()) {
             return false;
         }
-        UploadFieldDefinition other = (UploadFieldDefinition) obj;
-        return Objects.equals(name, other.name) && Objects.equals(required, other.required)
-                && Objects.equals(type, other.type);
+        UploadFieldDefinition that = (UploadFieldDefinition) o;
+        return required == that.required &&
+                Objects.equals(fileExtension, that.fileExtension) &&
+                Objects.equals(mimeType, that.mimeType) &&
+                Objects.equals(minAppVersion, that.minAppVersion) &&
+                Objects.equals(maxAppVersion, that.maxAppVersion) &&
+                Objects.equals(maxLength, that.maxLength) &&
+                Objects.equals(name, that.name) &&
+                type == that.type;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        return Objects.hash(fileExtension, mimeType, minAppVersion, maxAppVersion, maxLength, name, required, type);
+    }
+
+    /** {@inheritDoc} */
     @Override
     public String toString() {
-        return String.format("UploadFieldDefinition[name=%s, required=%b, type=%s]", name, required, type.name());
+        return "UploadFieldDefinition[" +
+                "fileExtension=" + fileExtension +
+                ", mimeType=" + mimeType +
+                ", minAppVersion=" + minAppVersion +
+                ", maxAppVersion=" + maxAppVersion +
+                ", maxLength=" + maxLength +
+                ", name=" + name +
+                ", required=" + required +
+                ", type=" + type.name() +
+                "]";
```

Comment:
I've started to use ToStringBuilder with Utilities.TO_STRING_STYLE, which produces this style in a pretty easy to maintain builder format, eg. new ToStringBuilder(this, TO_STRING_STYLE).append("name", name).build();